src/DynamoCoreWpf/ViewModels/Search/NodeAutoCompleteSearchViewModel.cs

```diff
@@ -75,9 +84,16 @@ internal IEnumerable<NodeSearchElement> GetMatchingNodes()
 
             foreach (var descriptor in functionDescriptors)
             {
-                if (descriptor.ReturnType.ToString() == inputPortType)
+                if ((descriptor.ReturnType.ToString() == inputPortType) || DerivesFrom(inputPortType, descriptor.ReturnType.ToString(), dynamoViewModel.Model.EngineController.LiveRunnerCore))
                 {
-                    elements.Add(new ZeroTouchSearchElement(descriptor));
+                    //TODO it's strange we create a new search element instead of using the existing ones in the model...
+                    var entry = new ZeroTouchSearchElement(descriptor);
```

Comment:
@QilongTang @aparajit-pratap I'm curious if there is a reason we create a new ZT searchElement instead of accessing the model and using the existing ZTSearchElements, this has some benefits, one of them being it's more consistent with the next few lines where NodeModelSearchElements are extracted from the Model.

Reply:
I'm not sure if it will be more work to search for an existing `ZTSearchElement` with the given descriptor as opposed to instantiating a new one.

Reply:
well, we don't need to search for anything if we use the loaded searchElements directly from the model instead of using function descriptors.

If that was the reason then - I'll make a small change to `ZTSearchElement` to expose the fully qualified output parameter name and I think we can unify this.

Reply:
Sure, that sounds like a good suggestion. I can't remember why I chose to use functiondescriptor but now that you point it out, it might just be easier to deal with search elements from the model.